test/core/end2end/tests/payload_with_proxy_auth.c

```diff
@@ -0,0 +1,302 @@
+  grpc_slice status_details = grpc_slice_from_static_string("xyz");
+  op->data.send_status_from_server.status_details = &status_details;
+  op->flags = 0;
+  op->reserved = NULL;
+  op++;
+  error = grpc_call_start_batch(s, ops, (size_t)(op - ops), tag(103), NULL);
+  GPR_ASSERT(GRPC_CALL_OK == error);
+
+  CQ_EXPECT_COMPLETION(cqv, tag(103), 1);
+  CQ_EXPECT_COMPLETION(cqv, tag(1), 1);
+  cq_verify(cqv);
+
+  GPR_ASSERT(status == GRPC_STATUS_OK);
+  GPR_ASSERT(0 == grpc_slice_str_cmp(details, "xyz"));
+  GPR_ASSERT(0 == grpc_slice_str_cmp(call_details.method, "/foo"));
+  validate_host_override_string("foo.test.google.fr:1234", call_details.host,
+                                config);
+  GPR_ASSERT(was_cancelled == 0);
+  GPR_ASSERT(byte_buffer_eq_slice(request_payload_recv, request_payload_slice));
+  GPR_ASSERT(
+      byte_buffer_eq_slice(response_payload_recv, response_payload_slice));
+
+  grpc_slice_unref(details);
+  grpc_metadata_array_destroy(&initial_metadata_recv);
+  grpc_metadata_array_destroy(&trailing_metadata_recv);
+  grpc_metadata_array_destroy(&request_metadata_recv);
+  grpc_call_details_destroy(&call_details);
+
+  grpc_call_unref(c);
+  grpc_call_unref(s);
+
+  cq_verifier_destroy(cqv);
+
+  grpc_byte_buffer_destroy(request_payload);
+  grpc_byte_buffer_destroy(response_payload);
+  grpc_byte_buffer_destroy(request_payload_recv);
+  grpc_byte_buffer_destroy(response_payload_recv);
+}
+
+/* Client sends a request with payload, server reads then returns a response
+   payload and status. */
+static void test_invoke_request_response_with_payload_and_proxy_auth(
+    grpc_end2end_test_config config) {
+  /* Indicate that the proxy requires user auth */
+  grpc_arg client_arg = {.type = GRPC_ARG_INTEGER,
+      .key = GRPC_END2END_HTTP_PROXY_TEST_CONNECT_AUTH_PRESENT,
+      .value.integer = 0};
+  grpc_channel_args client_args = {.num_args = 1, .args = &client_arg};
+  grpc_end2end_test_fixture f = begin_test(
+      config, "test_invoke_request_response_with_payload_and_proxy_auth",
+      &client_args, NULL);
+  request_response_with_payload_and_proxy_auth(config, f);
+  end_test(&f);
+  config.tear_down_data(&f);
+}
+
+static void test_invoke_10_request_response_with_payload_and_proxy_auth(
+    grpc_end2end_test_config config) {
+  int i;
+  /* Indicate that the proxy requires user auth */
+  grpc_arg client_arg = {.type = GRPC_ARG_INTEGER,
+      .key = GRPC_END2END_HTTP_PROXY_TEST_CONNECT_AUTH_PRESENT,
+      .value.integer = 0};
+  grpc_channel_args client_args = {.num_args = 1, .args = &client_arg};
+  grpc_end2end_test_fixture f = begin_test(
+      config, "test_invoke_10_request_response_with_payload_and_proxy_auth",
+      &client_args, NULL);
+  for (i = 0; i < 10; i++) {
+    request_response_with_payload_and_proxy_auth(config, f);
+  }
+  end_test(&f);
+  config.tear_down_data(&f);
+}
+
+void payload_with_proxy_auth(grpc_end2end_test_config config) {
```

Comment:
There's no point in running this test in any fixture except for h2_http_proxy_test.  So let's add a new FEATURE_MASK_* macro to indicate that the fixture uses an HTTP proxy, make sure that the h2_http_proxy_test fixture sets that bit, and then change the code here to only run the tests if that bit is set.